### PR TITLE
TINKERPOP-2391 Remove GLV templating system

### PR DESCRIPTION
**Pull request by spmallette**
https://issues.apache.org/jira/browse/TINKERPOP-2391

All tests pass with `docker/build.sh -t -n -i`

VOTE +1

### CODE REVIEW

**Comment by spmallette** [comment]
> Does updating the versions in the GLVs automatically by Maven still work with this change?

Thanks for pointing this out. Python will still auto-update, but it does not appear that .NET or Javascript will. I'd like to avoid version discrepancy so I think I will try to see what means we have to make these work as python does.

> (nitpick) Line 410 in the root pom.xml can also be removed now.

i'll remove that

> The generated files in the gremlin-dotnet module also all contain a comment mentioning the fact that they were generated and should not be edited by hand. We should remove this comment as they now can be edited by hand.

Another good point - thank you

**Comment by spmallette** [comment]
@FlorianHockmann please have a look at these latest round of changes - i think they address your concerns. I had to use the gmaven plugin to automatically update the project files for GLVs. there was no way i could see to do it with the maven versions plugin. the solution is a bit brittle because it just uses some regex to update the files on builds, but it shouldn't update all too often so perhaps it is ok. i can't think of how else to do it. i suppose i could have avoided regex since these were structured files but it didn't seem worth it given how fast/easy it was to just do the way i did it. please let me know what you think when you get a chance.

File: gremlin-dotnet/src/pom.xml [hunk 1]
```diff
@@ -33,14 +33,43 @@ limitations under the License.
...
+                        <version>${groovy.version}</version>
+                        <type>pom</type>
+                        <scope>runtime</scope>
+                    </dependency>
+                </dependencies>
                 <executions>
                     <execution>
-                        <id>generate-dsl</id>
-                        <phase/>
+                        <id>update-version</id>
```
[the remark this answers is not in the document]
Comment (spmallette) [reply gremlin-dotnet/src/pom.xml:50]: wow - i keep missing the template reference. thanks for noticing that. i suppose that my changes here get rid of the ability to do SNAPSHOT style releases to nuget. it doesn't appear as though we've ever done one and I'd venture to say that we never will. We would only ever do an "rc" version and we do that by way of a tag in git to capture the commit id from which we release that from (and we don't even bother to commit/push with the pom.xml updated to reflect the "rc"). 

I think that by not worrying about SNAPSHOTs we can do as you say and remove that "template" code in the "deploy" segment of the pom. In place of that "template" code I've added something to stop the build if the version is in SNAPSHOT. That approach does put it at odds with other GLVs which can support a form of SNAPSHOT but for now I don't think that's a problem.

File: gremlin-dotnet/src/pom.xml [hunk 2]
```diff
@@ -33,14 +33,43 @@ limitations under the License.
...
+                            <goal>execute</goal>
+                        </goals>
+                        <configuration>
+                            <scripts>
+                                <script>
+def mavenVersion = "${project.version}"
+def rawVersion = mavenVersion.replace("-SNAPSHOT", "")
+def file = new File("${project.basedir}/Gremlin.Net/Gremlin.Net.csproj")
+file.write(file.getText("UTF-8").replaceFirst(/&lt;Version&gt;(.*)&lt;\/Version&gt;/, "&lt;Version&gt;" + mavenVersion + "&lt;/Version&gt;").
+                                 replaceFirst(/&lt;FileVersion&gt;(.*)&lt;\/FileVersion&gt;/, "&lt;FileVersion&gt;" + rawVersion + ".0&lt;/FileVersion&gt;").
```
[the remark this answers is not in the document]
Comment (spmallette) [reply gremlin-dotnet/src/pom.xml]: thanks for the reference with that blog post. amazing how many "versions" .net has. wow. i opted to remove the two elements.

File: gremlin-dotnet/src/pom.xml [hunk 3]
```diff
@@ -33,14 +33,41 @@ limitations under the License.
...
+                        <id>update-version</id>
+                        <phase>generate-sources</phase>
+                        <goals>
+                            <goal>execute</goal>
+                        </goals>
+                        <configuration>
+                            <scripts>
+                                <script>
+def mavenVersion = "${project.version}"
+def rawVersion = mavenVersion.replace("-SNAPSHOT", "")
```
Comment (FlorianHockmann) [inline gremlin-dotnet/src/pom.xml:59]: Just a nit, but `rawVersion` isn't used anywhere so it can be removed here.

**Pull request merged**